lib/user.rb

```diff
@@ -0,0 +1,43 @@
+require_relative 'db_manager'
+
+class User
+
+  attr_reader :name, :username, :email, :password, :id
+
+  def initialize(name, username, email, password, id)
+    @name = name
+    @username = username
+    @email = email
+    @password = password
+    @id = id
+  end
+
+  def self.create(name, username, email, password)
+    result = DBManager.query("INSERT INTO users (name, username, email, password)
+                            VALUES ('#{name}', '#{username}', '#{email}', '#{password}')
+                            RETURNING id, name, username, email, password;"
+                            )
+    user_from_db_query(result[0])
+  end
+
+  def self.find(id)
+    result = DBManager.query("SELECT * FROM users WHERE id = '#{id}'")
+    user_from_db_query(result[0])
+  end
```

Comment:
This has given me great insight into how to begin to incorporate sign-in/sessions functionality on a website.